Add Getting Started with SelectBox (#1024)
DevExpress/devextreme-documentation: concepts/05 Widgets/SelectBox/00 Getting Started with SelectBox/25 Group Data.md

                        data: data,
                        type: 'array',
                        key: 'ID',
                    },
                    group: "Category"
                })
            }
        },
        // ...
    }
    </script>

##### React

    <!-- tab: App.js -->
    // ...
    import DataSource from 'devextreme/data/data_source';

    const dataSource = new DataSource({
        store: {
            data: data,
            type: 'array',
            key: 'ID'
        },
        group: "Category"
    })

    class App extends React.Component {
        // ...

        render() {
            return (
                <SelectBox ...
                    grouped={true}
                />
            );        
        }

    }

    export default App;
    
---

If your data is already grouped, ensure each group has the **key** and **items** fields as shown in [this article](/Documentation/Guide/Widgets/SelectBox/Grouping/In_the_Data_Source/). If the fields are named differently, implement the **DataSource**'s [map](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#map) function to create **key** + **items** field mappings.

> Run the code and ensure the widget's data is grouped.

Reviewer: Run code and ensure the widget's data is grouped.

is it and ensure or to ensure?

Author: and ensure.

Reviewer: ```suggestion
Run code and ensure widget data is grouped.
```sorrry, i skipped the second part of the sentence somehow last time

> The **SelectBox** can display data grouped by category. To implement this, we will use the data from the previous steps with the [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) component. Its API allows you to sort, filter, select, and group data. At its core, the **DataSource** has a [store](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/store/) - an object that keeps data and provides an API to access and modify it. To configure the store, use the **DataSource**'s [store](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/store/) object. Specify its [type](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/store/#type) as *"array"*, pass the initial data array to the [data](/Documentation/ApiReference/Data_Layer/ArrayStore/Configuration/#data) field, and set the [key](/Documentation/ApiReference/Data_Layer/ArrayStore/Configuration/#key) field. 

Reviewer 2: ```suggestion
The **SelectBox** can display data grouped by category. To implement this, we will use the data from the previous steps with the [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) component. Its API allows you to sort, filter, select, and group data. At its core, the **DataSource** has a [store](/Documentation/Guide/Data_Binding/Data_Layer/#Creating_DataSource/What_Are_Stores) - an object that keeps data and provides an API to access and modify it. To configure the store, use the **DataSource**'s [store](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/store/) object. Specify its [type](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/store/#type) as *"array"*, pass the initial data array to the [data](/Documentation/ApiReference/Data_Layer/ArrayStore/Configuration/#data) field, and set the [key](/Documentation/ApiReference/Data_Layer/ArrayStore/Configuration/#key) field. 
```

---
##### jQuery

    <!-- tab: index.js -->
    $(function() {
        $("#selectBox").dxSelectBox({
            const data = [{
                ID: 1,
                Name: 'Banana',
                Category: 'Fruits',
            }, {
                ID: 2,
                Name: 'Cucumber',
                Category: 'Vegetables',
            }, {
                ID: 3,
                Name: 'Apple',
                Category: 'Fruits',
            }, {
                ID: 4,
                Name: 'Tomato',
                Category: 'Vegetables',
            }, {
                ID: 5,
                Name: 'Apricot',
                Category: 'Fruits',
            }]

            const dataSource = new DevExpress.data.DataSource({
                store: data,
                type: 'array',
                key: "ID",
                group: "Category"
            });

            $("#selectBox").dxSelectBox({
                // ...

>                 onValueChanged: function(e) {
>                     if (e.previousValue !== null) {
>                         console.log(e.previousValue.Name);
>                     }
>                     console.log(e.value.Name);
>                 },

Reviewer 2: ```suggestion
```

                grouped: true
            });
        });
    });

##### Angular

    <!-- tab: app.component.html -->

>     <dx-select-box
>         // ...

Reviewer 2: ```suggestion
    <dx-select-box ...
```

        [grouped]="true">
    </dx-select-box>

    <!-- tab: app.component.ts -->
    // ...
    import DataSource from 'devextreme/data/data_source';

    @Component({
        selector: 'app-root',
        templateUrl: './app.component.html',
        styleUrls: ['./app.component.css']
    })
    export class AppComponent {
        data: Item[];
        dataSource: DataSource;

        constructor(service: AppService) {
            this.data = service.getItems();
            this.dataSource = new DataSource({
                store: {
                    data: this.data,
                    type: 'array',
                    key: 'ID'
                },
                group: "Category"
            });
        }

> 
>         onValueChanged(e) {
>             if (e.previousValue !== null) {
>                 console.log(e.previousValue.Name);
>             }
>             console.log(e.value.Name);
>         }

Reviewer 2: ```suggestion
```

    }

##### Vue

    <!-- tab: App.vue -->
    <template>

>         <DxSelectBox
>             // ...
>             @value-changed="onValueChange"

Reviewer 2: ```suggestion
        <DxSelectBox ...
```

            :grouped="true"
        />
    </template>

    <script>
    // ...
    import DataSource from 'devextreme/data/data_source';

    export default {
        components: {
            DxSelectBox
        },
        data() {
            return {
                dataSource: new DataSource({
                    store: {
                        data: data,
                        type: 'array',
                        key: 'ID',
                    },
                    group: "Category"
                })
            }
        },

>         methods: {
>             onValueChange(e) {
>                 if (e.previousValue) { 
>                     console.log(e.previousValue.Name); 
>                 }
>                 console.log(e.value.Name);
>             }
>         },

Reviewer 2: ```suggestion
        // ...
```

    }
    </script>

##### React

    <!-- tab: App.js -->
    // ...
    import DataSource from 'devextreme/data/data_source';

    const dataSource = new DataSource({
        store: {
            data: data,
            type: 'array',
            key: 'ID'
        },
        group: "Category"
    })

    class App extends React.Component {

>         onValueChanged(e) {
>             if (e.previousValue) { 
>                 console.log(e.previousValue.Name); 
>             }
>             console.log(e.value.Name);
>         }
> 

Reviewer 2: ```suggestion
        // ...
```

        render() {
            return (

>                 <SelectBox
>                     // ...
>                     onValueChanged={this.onValueChanged}

Reviewer 2: ```suggestion
                <SelectBox ...
```

                    grouped={true}
                />

> 

Reviewer 2: ```suggestion
```

> The **SelectBox** can display data grouped by category. To make this possible, use the [DataSource]() component. Its API allows you to [sort](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#sort), [filter](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#filter), [select](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#select), and [group](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/#group) data. At its core, the **DataSource** has a [store](/Documentation/ApiReference/Data_Layer/DataSource/Configuration/store/) - an object that keeps data and provides an API to access and modify it. To configure the store, use the **DataSource**'s [store]() object. Specify its [type]() as *"array"*, pass the initial data array to the [data]() field, and set the [key] field. 

Reviewer 2: Links to sort, filter, select, and group distract from the narrative. We better remove them. At the same time, the other links are missing.

        $("#selectBox").dxSelectBox({
            const data = [{
                ID: 1,
                Name: 'Banana',
                Category: 'Fruits',
            }, {
                ID: 2,
                Name: 'Cucumber',
                Category: 'Vegetables',
            }, {
                ID: 3,
                Name: 'Apple',
                Category: 'Fruits',
            }, {
                ID: 4,
                Name: 'Tomato',
                Category: 'Vegetables',
            }, {
                ID: 5,
                Name: 'Apricot',
                Category: 'Fruits',
            }]

            var dataSource = new DevExpress.data.DataSource({
                store: data,
                type: 'array',
                key: "ID",
                group: "Category"
            });

            $("#selectBox").dxSelectBox({
                dataSource: dataSource,
                displayExpr: "Name",
                searchEnabled: "true",
                onValueChanged: function(e) {
                    if (e.previousValue !== null) {
                        console.log(e.previousValue.Name);
                    }
                    console.log(e.value.Name);

>                 },

Reviewer 2: The only change is the `grouped` option. There's no need to show full SelectBox configuration.

                grouped: true
            });
        });
    });

##### Angular

    <!-- tab: app.component.html -->
    <dx-select-box
        [dataSource]="dataSource"
        displayExpr="Name"

>         searchEnabled="true"

Reviewer 3: ```suggestion
        [searchEnabled]="true"
```

        (onValueChanged)="onValueChanged($event)"

>         grouped="true">

Reviewer 3: ```suggestion
        [grouped]="true">
```

                }
                console.log(e.value.Name);
            }
        },
    }
    </script>

##### React

    <!-- tab: App.js -->
    import React from 'react';
    import 'devextreme/dist/css/dx.common.css';
    import 'devextreme/dist/css/dx.light.css';
    import { SelectBox } from 'devextreme-react/select-box';
    import DataSource from 'devextreme/data/data_source';
    import { data } from './data';

    const dataSource = new DataSource({
        store: {
            data: data,
            type: 'array',
            key: 'ID'
        },
        group: "Category"
    })

    class App extends React.Component {
        onValueChanged(e) {
            if (e.previousValue) { 
                console.log(e.previousValue.Name); 
            }
            console.log(e.value.Name);
        }

        render() {
            return (
                <SelectBox
                    dataSource={dataSource}
                    displayExpr="Name"
                    searchEnabled={true}
                    onValueChanged={this.onValueChanged}
                    grouped={true}
                />

            );        
        }

    }

    export default App;
    
---

> If your data is already grouped, ensure each group has the **key** and **items** fields as shown in [this article](/Documentation/Guide/Widgets/SelectBox/Grouping/In_the_Data_Source/). If the fields are named differently, implement the **DataSource**'s [map]() function to create **key** + **items** field mappings.

Reviewer 2: The link to `map` is missing.
